front/lib/front.js

```diff
@@ -18,22 +18,19 @@ export function useRegisterUnloadHandlers(
 
   // Add handler for next.js router events that don't load a new page in the browser.
   const router = useRouter();
-  useEffect(
-    (e) => {
-      const confirmBrowseAway = () => {
-        if (!editorDirty) return;
-        if (window.confirm(unloadWarning)) return;
+  useEffect(() => {
+    const confirmBrowseAway = () => {
+      if (!editorDirty) return;
+      if (window.confirm(unloadWarning)) return;
 
-        router.events.emit("routeChangeError");
-        throw "routeChange aborted.";
-      };
+      router.events.emit("routeChangeError");
+      throw "routeChange aborted.";
```

Comment:
not your code sorry 😬  But should we change this to `throw new Error(...)` ? Throwing just a string is not ideal

Reply:
:+1: (for context: this code was written by an external contributor a long time ago)

Reply:
Note I tested making it a `new Error(...)` and it changes the behavior in dev (pops up an error panel). Not sure what it would do in production.

Right now the error is unhandled and ends up in the console.log which is hacky at best. Let's revisit later.